src/lib/dom.ts

```diff
@@ -668,34 +668,61 @@ export function anchors(includeInvisible = false) {
     ])
 }
 
+export const enum TabTarget {
+    CurrentTab,
+    NewTab,
+    NewBackgroundTab,
+    NewWindow,
+}
+
+const tabTargetToModifierKey = {
+    [TabTarget.CurrentTab]: {},
+    [TabTarget.NewTab]: { ctrlKey: true, shiftKey: true },
+    [TabTarget.NewBackgroundTab]: { ctrlKey: true },
+    [TabTarget.NewWindow]: { shiftKey: true },
+}
+
 /** if `target === _blank` clicking the link is treated as opening a popup and is blocked. Use webext API to avoid that. */
-export function simulateClick(target: HTMLElement) {
+export function simulateClick(
+    target: HTMLElement,
+    tabTarget: TabTarget = TabTarget.CurrentTab,
+) {
     // target can be set to other stuff, and we'll fail in annoying ways.
     // There's no easy way around that while this code executes outside of the
     // magic 'short lived event handler' context.
     //
     // OTOH, hardly anyone uses that functionality any more.
-    if (
+    let usePopupBlockerWorkaround =
         (target as HTMLAnchorElement).target === "_blank" ||
         (target as HTMLAnchorElement).target === "_new"
-    ) {
+    const href = (target as HTMLAnchorElement).href
+    if (href !== undefined && href.startsWith("file:")) {
```

Comment:
You can simplify this to

```suggestion
    if (href?.startsWith("file:")) {
```

Reply:
Thanks for the tip :)